dafni_cli/upload.py

```diff
@@ -0,0 +1,91 @@
+import click
+from click import Context
+from click.testing import CliRunner
+from pathlib import Path
+from requests.exceptions import HTTPError
+from requests import Response
+
+from dafni_cli.login import check_for_jwt_file
+from dafni_cli.api.models_api import (
+    validate_model_definition,
+    get_model_upload_urls,
+    upload_file_to_minio,
+    model_version_ingest
+)
+from dafni_cli.utils import argument_confirmation
+
+
+@click.group()
+@click.pass_context
+def upload(ctx: Context):
+    """Uploads entities (models, datasets, workflows, groups) to DAFNI.
+
+    Args:
+        ctx (Context): Context containing JWT of the user.
+    """
+    ctx.ensure_object(dict)
+    jwt_dict, _ = check_for_jwt_file()
+    ctx.obj["jwt"] = jwt_dict["jwt"]
+
+
+@upload.command()
+@click.argument("definition", nargs=1, required=True, type=click.Path(exists=True))
+@click.argument("image", nargs=1, required=True, type=click.Path(exists=True))
+@click.option("--version-message", nargs=1, required=True, type=str)
+@click.option("--parent-model", type=str, default=None)
+@click.pass_context
+def model(
+        ctx: Context, definition: click.Path, image: click.Path, version_message: str, parent_model: str
```

Comment:
I would recommend knocking up a simple utils function to validate the paths are actually valid before you do anything with them. something like `os.path.exists(path)`, are we ok with relative paths etc?

Reply:
So Click already has a pretty useful error for this:
<pre>
Usage: runner.py upload model [OPTIONS] DEFINITION IMAGE
Try 'runner.py upload model --help' for help.

Error: Invalid value for 'DEFINITION': Path 'model-file-examples\no_definition.yaml' does not exist.`
</pre>

This comes from the `exists=True` argument. Do you think we need to do a check as well?

Reply:
ah ok, hadn't come across that. In that case, nope that should suffice